src/plugins/abrt-journal.c

```diff
@@ -469,15 +469,16 @@ void abrt_journal_watch_notify_strings(abrt_journal_watch_t *watch, void *data)
         error_msg_and_die("Cannot read journal data.");
 
     GList *cur = conf->strings;
-    while (cur)
-    {
+    for (; cur; cur = g_list_next(cur))
         if (strstr(message, cur->data) != NULL)
             break;
 
-        cur = g_list_next(cur);
-    }
+    GList *blacklist_cur = conf->blacklisted_strings;
```

Comment:
With no white-list match, there is no need to iterate over the black-list items.